From 6d1caaf74b90be2ccb6e725a1ed1551d2b564e4b Mon Sep 17 00:00:00 2001
From: Sanyog Kale <skale@habana.ai>
Date: Thu, 10 Nov 2022 11:36:08 +0200
Subject: [PATCH] Revert "Revert "[SW-104919] gaudi2: security: fix for ppboot hang""

This reverts commit 848e4355cb0782de15a657fbf481cd5aa889f477.

It is observed with certain setups, pcie link is not established
on sudo reboot with ARC0 using its in own DCCM. Revert this change
till issue is analyzed.

Change-Id: I9894e51bb99f1d316543a4c8d4cb08c6a7c57aaf
---

diff --git a/arch/gaudi2/arc/common.c b/arch/gaudi2/arc/common.c
index 70e6402..28b322c 100644
--- a/arch/gaudi2/arc/common.c
+++ b/arch/gaudi2/arc/common.c
@@ -336,13 +336,16 @@
 	return 0;
 }
 
+/* TODO: This does not need to be global. Make it local when
+ * https://jira.habana-labs.com/browse/SW-105874 is concluded.
+ */
+static uint32_t magic;
 static int ppboot_bringup_preboot(u8 image_ptr)
 {
 	u32 image_size, payload_size, copy_size;
 	u32 flash_offset;
 	u32 num_cycles;
 	int ret;
-	uint32_t magic = 0;
 	uint8_t hash[SHA384_SUM_LEN] = { 0 };
 
 	flash_offset = ppboot_flash_image_offset(image_ptr);
